### PR TITLE
Fix issue where StepInfo config could be parsed into a Step

### PR DESCRIPTION
**Pull request by lgatys**
Changes proposed in this pull request:

- Prevents `construct_arg` method in `from_params.py` to try to parse the argument as a Step, if it is the `config` argument of the `StepInfo` class. 
- This happened when the `config` contained references to upstream steps, which evaluated the `could_be_step` condition to `True`.
- The previous behavior was creating issues with the WandbWorkspace when trying to update and re-parse the StepInfo.

## Before submitting

- [x] I've read and followed all steps in the [Making a pull request](https://github.com/allenai/tango/blob/main/CONTRIBUTING.md#making-a-pull-request)
    section of the `CONTRIBUTING` docs.
- [x] If this PR fixes a bug, I've added a test that will fail without my fix.

## After submitting

- [ ] All GitHub Actions jobs for my pull request have passed.

### CODE REVIEW

File: tests/step_info_test.py
Comment (epwalsh) [inline tests/step_info_test.py:36]:
```suggestion
        def run(self, bar: Any) -> str:  # type: ignore
```

File: tests/step_info_test.py
Comment (epwalsh) [inline tests/step_info_test.py:41]:
```suggestion
        def run(self) -> str:  # type: ignore
```

**Review by epwalsh — COMMENTED** [review]
Just a couple minor comments!

File: tests/step_info_test.py
Comment (epwalsh) [reply tests/step_info_test.py:36]: This will tell `mypy` to stop complaining the "Lint" checks.

File: tests/step_info_test.py
Comment (lgatys) [reply tests/step_info_test.py]: Done!

**Review by epwalsh — APPROVED** [review]
LGTM! I'll make a patch release once this merges

**Pull request merged**
